### PR TITLE
add an option to turn off tendencies

### LINKED ISSUE #2863: Add prognostic edmf diffusion only aquaplanet simulation

### PR DESCRIPTION
**Pull request by szy21**
## Purpose 

Adds an option to turn off grid-scale or subgrid-scale tendencies for edmf tests. I didn't add an option to turn off both tendencies as I don't think that's very useful. Also adds a prognostic edmf diffusion only test.

Closes #2863 

## Content

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

File: .buildkite/longruns_gpu/pipeline.yml
Comment (szy21) [inline .buildkite/longruns_gpu/pipeline.yml:1]: I'll revert the changes after testing.

File: src/prognostic_equations/zero_tendency.jl [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    t,
+    colidx,
+    ::NoSubgridScaleTendency,
+    ::PrognosticEDMFX,
+)
+    # turn off all subgrid-scale tendencies
+    n = n_prognostic_mass_flux_subdomains(p.atmos.turbconv_model)
+    Yₜ.c.sgsʲs[colidx] .= zero(eltype(Yₜ.c.sgsʲs[colidx]))
+    Yₜ.f.sgsʲs[colidx] .= zero(eltype(Yₜ.f.sgsʲs[colidx]))
+    # for j in 1:n
```
Comment (trontrytel) [inline src/prognostic_equations/zero_tendency.jl:30]: Is this still needed?

File: src/solver/model_getters.jl [hunk 2]
```diff
@@ -438,3 +438,15 @@ function get_tracers(parsed_args)
...
+
+function get_tendency_model(parsed_args)
+    zero_tendency_name = parsed_args["zero_tendency"]
+    @assert zero_tendency_name in (nothing, "grid_scale", "subgrid_scale")
+    return if zero_tendency_name == "grid_scale"
+        NoGridScaleTendency()
+    elseif zero_tendency_name == "subgrid_scale"
+        NoSubgridScaleTendency()
+    elseif isnothing(zero_tendency_name)
+        AllTendency()
```
Comment (trontrytel) [inline src/solver/model_getters.jl:450]: Maybe `UseAllTendecy` or something along those lines? `AllTendency` could be interpreted as zero all tendecies

File: src/prognostic_equations/zero_tendency.jl [hunk 3]
```diff
@@ -0,0 +1,37 @@
+###
+### Zero grid-scale or subgrid-scale tendencies
+###
+
+function zero_tendency!(Yₜ, Y, p, t, colidx, ::NoGridScaleTendency, _)
```
Comment (trontrytel) [inline src/prognostic_equations/zero_tendency.jl:5]: Why dont we also need the `nothing` version?

File: src/prognostic_equations/zero_tendency.jl (hunk 1 above)
Comment (szy21) [reply src/prognostic_equations/zero_tendency.jl:30]: This is testing gpu. I’ll see what works.

File: src/prognostic_equations/zero_tendency.jl (hunk 3 above)
Comment (szy21) [reply src/prognostic_equations/zero_tendency.jl:5]: I forgot to add it, thanks!

**Comment by szy21** [comment]
@trontrytel I addressed your comments and gave up on making it work on GPU - would you like to take a look again?

**Review by trontrytel — APPROVED** [review]
Thank you!

**Pull request merged**